docs/design/proposals/rbd-volume-healer.md

```diff
@@ -0,0 +1,94 @@
+# RBD NBD VOLUME HEALER
+
+- [RBD NBD VOLUME HEALER](#rbd-nbd-volume-healer)
+  - [Rbd Nbd](#rbd-nbd)
+    - [Advantages of userspace mounters](#advantages-of-userspace-mounters)
+    - [Side effects of userspace mounters](#side-effects-of-userspace-mounters)
+  - [Volume Healer](#volume-healer)
+    - [More thoughts](#more-thoughts)
+
+## Rbd Nbd
+
+The RBD CSI plugin will provision new RBD images and attach and mount those
+to workloads. Currently, the default mounter is krbd, which uses the kernel
+rbd driver to mount the rbd images onto the application node. Here on
+at cephcsi we will also have a userspace way of mounting the RBD images,
+via RBD-NBD.
+
+[Rbd-nbd](https://docs.ceph.com/en/latest/man/8/rbd-nbd/) is a client for
+RADOS block device (rbd) images like the existing rbd kernel module. It
+will map an rbd image to an NBD (Network Block Device) device, allowing
+access to it as a regular local block device.
+
+![csi-rbdnbd](https://user-images.githubusercontent.com/12432241/125400940-23e75d00-e3d0-11eb-806c-bab09318de54.png)
+
+It’s worth making a note that the rbd-nbd processes will run on the
+client-side, which is inside the `csi-rbdplugin` node plugin.
+
+### Advantages of userspace mounters
+
+- Adding changes to client logic to accommodate new features will be easy
+- Container upgrades will be independent of the host node, which means if
```

Comment:
rebooting is not needed now either, we expect `rbd.ko` to be available on the host, just like `nbd.ko`.

Reply:
If that confused you, please feel free to suggest a betterment, I mean to say:

If there are any new features in the rbd, for example, RBD_FEATURE_LAYERING, RBD_FEATURE_STRIPINGV2, RBD_FEATURE_EXCLUSIVE_LOCK, RBD_FEATURE_OBJECT_MAP, RBD_FEATURE_FAST_DIFF, RBD_FEATURE_DEEP_FLATTEN, etc

One has to implement this in the Linux kernel rbd driver and wait for the feature to be available in the kernel, once they are available one has to upgrade the kernel and reboot the host node. Thankfully this is not needed for userspace mounters as we get them in userspace via librbd without needing to reboot the node.

Reply:
sure, I get that part, but that is just repeating the previous point? 'container upgrades' would be ceph-csi new version, I guess, so mentioning something about new rbd features might help.